codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -282,7 +282,7 @@ private int allocateBytesForTree(Http2Stream parent, int connectionWindow) {
                 int weight = child.weight();
                 double weightRatio = weight / (double) totalWeight;
 
-                int bytesForTree = Math.min(nextConnectionWindow, (int) Math.ceil(connectionWindow * weightRatio));
+                int bytesForTree = Math.min(nextConnectionWindow, Math.max(1, (int) (connectionWindow * weightRatio)));
```

Comment:
@nmittler - I think we need some form of rounding to ensure progress is made.  I'm fine with this.  Perhaps we should document our relaxed "requirements" for distributing priority now?  Something like `priority is generally distributed proportionally to siblings but the rounding remainders are indescrimenetly distributed.`